### PR TITLE
fixes windoor smartglassing

### LINKED ISSUE #16128: toggling transparency on an open windoor results in an opaque, open windoor
(WEB REPORT BY: sonixapache REMOTE: 172.93.109.202:7777)
# Revision
f15c6455f1884417d22e5479aa89bdc42ab8e30a

# Description
toggling transparency on an open windoor results in an opaque, open windoor

need to fix this myself asap

### PR DESCRIPTION
**Pull request by SonixApache**
resolves #16128

🆑 
 - bugfix: unfucked windoor smartglassifying

### CODE REVIEW

File: code/game/machinery/doors/windowdoor.dm [hunk 1]
```diff
@@ -23,7 +23,7 @@
 	penetration_dampening = 2
 	animation_delay = 7
 	var/obj/machinery/smartglass_electronics/smartwindow
-	var/was_opaque
+	var/window_is_opaque = 0 //The var that helps darken the glass when the door opens/closes
```
Comment (N3X15) [inline code/game/machinery/doors/windowdoor.dm]: Use FALSE

File: code/game/machinery/doors/windowdoor.dm [hunk 2]
```diff
@@ -39,12 +39,20 @@
...
 		animate(src, color="#FFFFFF", time=5)
-		set_opacity(0)
 	else
 		animate(src, color="#222222", time=5)
-		set_opacity(1)
+	
+	if(density) //window is CLOSED
+		if(window_is_opaque) //Is it dark?
+			set_opacity(0) //Make it light.
+			window_is_opaque = 0
```
Comment (N3X15) [inline code/game/machinery/doors/windowdoor.dm]: FALSE

File: code/game/machinery/doors/windowdoor.dm [hunk 3]
```diff
@@ -39,12 +39,20 @@
...
+		else
+			set_opacity(1) // Else, make it dark.
+			window_is_opaque = 1
```
Comment (N3X15) [inline code/game/machinery/doors/windowdoor.dm]: TRUE

**Review by N3X15 — CHANGES_REQUESTED** [review]

File: code/game/machinery/doors/windowdoor.dm (hunk 3 above)
Comment (SonixApache) [reply code/game/machinery/doors/windowdoor.dm]: done

File: code/game/machinery/doors/windowdoor.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/machinery/doors/windowdoor.dm]: done

File: code/game/machinery/doors/windowdoor.dm (hunk 2 above)
Comment (SonixApache) [reply code/game/machinery/doors/windowdoor.dm]: done

**Review by N3X15 — APPROVED** [review]

**Comment by N3X15** [comment]
Capitalize, literally `TRUE` and `FALSE`

**Comment by SonixApache** [comment]
done

**Pull request merged**